text/0000-forwardpropstoheader.md

```diff
@@ -0,0 +1,186 @@
+        screen: MainScreen,
+    }, SlaveScreen: {
+        screen: SlaveScreen
+    }
+}, {
+    headerMode: 'float',
+    onTransitionEnd: () => {
+        db.setTransitionEnded(true)
+    },
+})
+// state.js
+export class State {
+    transitionEnded = false
+    headerProps = null
+    setTransitionEnded(transitionEnded) {
+        this.transitionEnded = transitionEnded
+    }
+    setCurrentHeader(header) {
+        this.currentHeader = header
+    }
+    setHeaderProps(headerProps) {
+        this.headerProps = headerProps
+    }
+    slaveScreen = {
+        triggerEditMode: () => null,
+        setEditModeCallback: function(callback) {
+            this.triggerEditMode = callback
+        },
+    }
+}
+export let db: State = new State()
+// screens/SlaveScreen.js
+export default class CatchUpdateWrapper extends React.Component {
+    constructor(props) {
+        super(props)
+        this.state = { __updated_props: {} }
+    }
+
+    componentDidMount() {
+        db.setCurrentHeader(this)
+    }
+
+    componentWillUnmount() {
+        db.setCurrentHeader(null)
+    }
+
+    render() {
+        return <this.props.innerComponent {...this.props} {...this.state.__updated_props}/>
+    }
+}
+
+class HeaderProxy extends React.Component {
+    render() {
+        return null
+    }
+
+    componentDidUpdate() {
+        if (db.currentHeader !== null && db.transitionEnded) {
+            db.currentHeader.setState({
+                __updated_props: this.props,
+            })
+        }
+    }
+}
+export class SlaveScreen extends React.Component {
+    static navigationOptions = {
+        header: (props: *) => {
+            db.setHeaderProps(props)
+            return <CatchUpdateWrapper innerComponent={CustomHeader}
+                                       hideDuringTransition={true} {...props} />
+        },
+        headerBackTitle: null,
+        headerTruncatedBackTitle: null,
+        headerRight: <Button title={"Edit"} onPress={() => {
+            db.slaveScreen.triggerEditMode()
+        }}/>,
+    }
+
+    constructor(props) {
+        super(props)
+        this.state = { editMode: false }
+        db.slaveScreen.setEditModeCallback(this.triggerEditMode.bind(this))
+    }
+
+    triggerEditMode() {
+        this.setState({ editMode: true })
+    }
+
+    render() {
+        return (
+            <View style={{ backgroundColor: '#000', width: 1000, height: 1000 }}>
+                <HeaderProxy
+                    onChangeText={console.warn}
+                    hideDuringTransition={true}  // residue of not cleaned up CustomHeader class
+                    editStyle={this.state.editMode}
+                    {...db.headerProps}
+                />
+                <Text style={{ color: "#FFF" }}>I'm the body!</Text>
+            </View>
+        )
+    }
+}
+```
+
+You can try a working example with this: (uses Expo, it works when ejected too)
+
+```bash
+git clone https://github.com/CrazyPython/reactive-navigation-demo.git
```

Comment:
I get "Can't find variable db" after pressing "Go" when I run this example

Reply:
I will investigate, probably forgot to stage some file